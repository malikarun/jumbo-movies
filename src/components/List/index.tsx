import GridList from '@material-ui/core/GridList';
import GridListTile from '@material-ui/core/GridListTile';
import GridListTileBar from '@material-ui/core/GridListTileBar';
import IconButton from '@material-ui/core/IconButton';
import ListSubheader from '@material-ui/core/ListSubheader';
import { withStyles } from '@material-ui/core/styles';
import InfoIcon from '@material-ui/icons/Info';
import React, { Component, Fragment, ChangeEvent } from 'react';
import { Link } from 'react-router-dom';
import { IMedia } from '../../interfaces';
import TextField from '@material-ui/core/TextField';

interface IProps {
    header: string;
    path: string;
    items: IMedia[];
    classes: any;
}

interface IState {
    searchTerm: string;
}

const styles = {
    icon: {
        color: 'rgba(255, 255, 255, 0.54)',
    }
}

const posterUrl = 'https://image.tmdb.org/t/p/w300_and_h450_bestv2/';

class List extends Component<IProps, IState> {
    state = {
        searchTerm: ''
    }

    handleChange = (event: any) => {
        this.setState({ searchTerm: event.target.value});
    }

    render() {
        const { classes, items: terms, header, path } = this.props;
        const { searchTerm } = this.state;
        const field = path === 'movie' ? 'title' : 'name';
        const items = terms.filter((term: IMedia) => term[field].toLowerCase().includes(searchTerm.toLowerCase()));

        return (
            <Fragment>
                <TextField
                    label="Search"
                    placeholder="Placeholder"
                    fullWidth
                    onChange={this.handleChange}
                    InputLabelProps={{
                        shrink: true,
                    }}
                />

                <GridList cellHeight={180}>
                    <GridListTile key="Subheader" cols={2} style={{ height: 'auto' }}>
                        <ListSubheader component="div">{header}</ListSubheader>
                    </GridListTile>

                    {items && items.map((item: IMedia) => (
                        <GridListTile key={item.id}>
                            <img src={`${posterUrl}${item.poster_path}`} alt={item[field]} />

                            <GridListTileBar
                                title={item[field]}
                                subtitle={<span>Votes: {item.vote_count}</span>}
                                actionIcon={
                                    <Link to={`/${path}/${item.id}`}>
                                        <IconButton className={classes.icon}>
                                            <InfoIcon />
                                        </IconButton>
                                    </Link>
                                }
                            />
                        </GridListTile>
                    ))}
                </GridList>
            </Fragment>
        );
    }
}

export default withStyles(styles)(List);
