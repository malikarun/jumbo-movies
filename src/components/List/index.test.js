import React from 'react';
import List from './index';
import renderer from 'react-test-renderer';

test('Should create a list of items', () => {
    const items = [];

    const component = renderer.create(
        <List header="haha" path="tv" items={items}></List>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});