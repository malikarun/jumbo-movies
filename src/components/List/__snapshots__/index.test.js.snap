// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create a list of items 1`] = `
Array [
  <div
    className="MuiFormControl-root-2 MuiFormControl-fullWidth-5"
  >
    <label
      className="MuiFormLabel-root-17 MuiInputLabel-root-6 MuiInputLabel-formControl-11 MuiInputLabel-animated-14 MuiInputLabel-shrink-13"
      data-shrink={true}
    >
      Search
    </label>
    <div
      className="MuiInputBase-root-37 MuiInput-root-24 MuiInput-underline-28 MuiInputBase-fullWidth-46 MuiInput-fullWidth-31 MuiInputBase-formControl-38 MuiInput-formControl-25"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input-47 MuiInput-input-32"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Placeholder"
        required={false}
        type="text"
      />
    </div>
  </div>,
  <ul
    className="MuiGridList-root-54"
    style={
      Object {
        "margin": -2,
      }
    }
  >
    <li
      className="MuiGridListTile-root-55"
      style={
        Object {
          "height": "auto",
          "padding": 2,
          "width": "100%",
        }
      }
    >
      <div
        className="MuiGridListTile-tile-56"
      >
        <div
          className="MuiListSubheader-root-59 MuiListSubheader-sticky-64 MuiListSubheader-gutters-62"
        >
          haha
        </div>
      </div>
    </li>
  </ul>,
]
`;
